src/App.js

```diff
@@ -1,6 +1,97 @@
-import React from 'react';
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Posts } from './components/Posts';
+import { Pagination } from './components/Pagination';
+import { SelectSettings } from './components/SelectSettings';
 import './App.css';
 
-const App = () => <h1>Pagination</h1>;
+class App extends Component {
+  state = {
+    posts: [],
+    currentPage: 1,
+    postsPerPage: 10,
+  }
+
+  componentDidMount() {
+    axios.get(`https://jsonplaceholder.typicode.com/posts`)
+      .then((res) => {
+        const posts = res.data;
+
+        this.setState({ posts });
+      });
+  }
+
+  changePage = ({ target }) => {
+    const { currentPage } = this.state;
+
+    if (target.id === 'prev') {
+      currentPage !== 1
+        ? (this.setState(state => ({
+          currentPage: state.currentPage - 1,
+        })))
+        : (this.setState({
+          currentPage: 1,
+        }));
+    } else if (target.id === 'next') {
+      currentPage < +target.parentElement.id
```

Comment:
when you use parentElement you interact with dom, don't do this in React, there is an option to use arguments for functions